src/lib/y2partitioner/dialogs/partition_size.rb

```diff
@@ -10,143 +13,335 @@ module Dialogs
+        HVSquash(SizeWidget.new(@ptemplate, @regions))
+      end
+
+      # Like CWM::RadioButtons but each RB has a subordinate indented widget.
+      # This is kind of like Pager, but all Pages being visible at once,
+      # and enabled/disabled.
+      # Besides {#items} you need to define also {#widgets}.
+      class ControllerRadioButtons < CWM::CustomWidget
+        def initialize
+          self.handle_all_events = true
+        end
+
+        # @return [WidgetTerm]
+        def contents
+          Frame(
+            label,
+            HBox(
+              HSpacing(hspacing),
+              RadioButtonGroup(Id(widget_id), buttons_with_widgets),
+              HSpacing(hspacing)
+            )
+          )
+        end
+
+        # @return [Numeric] margin at both sides of the options list
+        def hspacing
+          1.45
+        end
+
+        # @return [Numeric] margin above, between, and below the options
+        def vspacing
+          0.45
+        end
+
+        # @return [Array<Array(id,String)>]
+        abstract_method :items
+
+        # FIXME: allow {WidgetTerm}
+        # @return [Array<AbstractWidget>]
+        abstract_method :widgets
+
+        # @param event [Hash] UI event
+        def handle(event)
+          eid = event["ID"]
+          ids.zip(widgets).each do |id, widget|
+            if id == eid
+              widget.enable
+            else
+              widget.disable
+            end
+          end
+          nil
+        end
+
+        # Get the currently selected radio button from the UI
+        def value
+          Yast::UI.QueryWidget(Id(widget_id), :CurrentButton)
+        end
+
+        # Tell the UI to change the currently selected radio button
+        def value=(val)
+          Yast::UI.ChangeWidget(Id(widget_id), :CurrentButton, val)
+        end
+
+        # @return [AbstractWidget] the widget corresponding
+        #   to the currently selected option
+        def current_widget
+          idx = ids.index(value)
+          widgets[idx]
+        end
+
+      private
+
+        # @return [Array<id>]
+        def ids
+          @ids ||= items.map(&:first)
+        end
+
+        def buttons_with_widgets
+          items = self.items
+          widgets = self.widgets
+          raise ArgumentError unless items.size == widgets.size
+
+          terms = items.zip(widgets).map do |(id, text), widget|
+            VBox(
+              VSpacing(vspacing),
+              Left(RadioButton(Id(id), Opt(:notify), text)),
+              Left(HBox(HSpacing(4), VBox(widget)))
+            )
+          end
+          VBox(*terms, VSpacing(vspacing))
+        end
       end
 
-      # Choose partition size: maximum, size based, start-end based.
-      class SizeWidget < CWM::CustomWidget
+      # Choose a size (region, really) for a new partition
+      # from several options: use maximum, enter size, enter start+end
+      class SizeWidget < ControllerRadioButtons
         include Yast
```

Comment:
why include of whole Yast is needed? later I see that Builtin are prefixed properly.